### PR TITLE
Amam/add subscription manager dp2p

### PR DESCRIPTION
**Pull request by mustofa-binary**
- pass subscribe method from subscription manager in binary static to dp2p
- pass client local currency properties to dp2p

### CODE REVIEW

File: src/javascript/app/pages/cashier/dp2p.js [hunk 1]
```diff
@@ -61,13 +62,28 @@ const DP2P = (() => {
                 `;
                 el_main_css.rel = 'stylesheet';
 
+                const p2pSubscribe = (request, cb) => {
+                    SubscriptionManager.subscribe(Object.keys(request)[0], request, cb);
```
Comment (negarn) [inline src/javascript/app/pages/cashier/dp2p.js]: can you add a comment about how request object should have the first param be the msg_type?

File: src/javascript/app/pages/cashier/dp2p.js [hunk 2]
```diff
@@ -61,13 +62,28 @@ const DP2P = (() => {
...
+                    return {
+                        unsubscribe: () => SubscriptionManager.forget(Object.keys(request)[0]),
+                    };
+                };
+
+                const binary_websocket = {
```
Comment (negarn) [inline src/javascript/app/pages/cashier/dp2p.js]: maybe we can rename it to just websocket_api?

File: src/javascript/app/pages/cashier/dp2p.js (hunk 1 above)
Comment (mustofa-binary) [reply src/javascript/app/pages/cashier/dp2p.js]: sure

File: src/javascript/app/pages/cashier/dp2p.js [hunk 3]
```diff
@@ -59,29 +66,43 @@ const DP2P = (() => {
...
+                is_virtual           : Client.get('is_virtual'),
+                local_currency_config: Client.get('local_currency_config'),
+                residence            : Client.get('residence'),
+            },
+            lang: getLanguage(),
+            websocket_api,
+        };
+
+        ReactDOM.render(
+            // eslint-disable-next-line no-console
```
Comment (mahdiaryayi-fs) [inline src/javascript/app/pages/cashier/dp2p.js]: Why we need this line (eslint-disable) in PR?

Comment (negarn) [reply src/javascript/app/pages/cashier/dp2p.js]: @mahdipourziaei-binary could you remove this line?

Comment (mahdipourziaei-binary) [reply src/javascript/app/pages/cashier/dp2p.js]: @negarn It's done. Just waiting for @mustofa-binary to get my commits.

Comment (mustofa-binary) [reply src/javascript/app/pages/cashier/dp2p.js]: its done 👍

**Pull request merged**